src/docker/dockerclient.cc

```diff
@@ -13,7 +13,8 @@ DockerClient::HttpClientFactory DockerClient::DefaultHttpClientFactory = []() {
 DockerClient::DockerClient(std::shared_ptr<HttpInterface> http_client) : http_client_{std::move(http_client)} {}
 
 bool DockerClient::serviceRunning(const std::string& app, const std::string& service, const std::string& hash) {
-  updateContainerStatus();
+  boost::mutex::scoped_lock lock(lock_);
```

Comment:
there is no need for thread sync here since an update cycle happens within a single thread.

Reply:
the lock is protecting the access to the root_; if there is no other thread in the system the lock can be removed. but assuming these things is _always_ a bad recipe bearing in mind that the architecture might evolve and that properly protecting variables cost nothing, just knowledge....